### DISCUSSION TITLE
Trend arrow to use multiple readings to smooth direction

### DISCUSSION THREAD

**Discussion by mattherasymiuk** [NightscoutFoundation/xDrip#2649] [Ideas]
The trend arrow looks like it uses the difference between the most recent and second most recent blood glucose readings to determine which direction to point. I often find if data gets noisy the trend arrow can jump up and down dramatically over the course of just a few readings (double arrow straight down, to single arrow straight up, to level arrow, etc.). 

Is there a way to use perhaps a few more past readings to determine trend arrow direction? This may help smooth out potentially jumpy readings and be a little more aesthetically pleasing.

**Comment by Navid200** [comment]
This comes up a lot.  It's either the trend arrow or the delta.

There is no way to do what you are suggesting without accepting some delay.  What you are suggesting is basically a low-pass filter.  A low-pass filter causes delay by definition.

The problem is that it will also delay a real sudden change.  There are individuals who prefer to see a sudden change as soon as possible even if the price to pay is visual fluctuations when the readings are noisy.
